plugins/GSdx/GSState.cpp

```diff
@@ -405,9 +405,27 @@ GSVideoMode GSState::GetVideoMode()
 	return videomode;
 }
 
-GSVector4i GSState::GetDisplayRect(int i, bool merged_rect)
+GSVector4i GSState::GetDisplayRect(int i)
 {
-	if(i < 0) i = IsEnabled(1) ? 1 : 0;
+	if (!IsEnabled(0) && !IsEnabled(1))
+		return GSVector4i(0);
+
+	// If no specific context is requested then pass the merged rectangle as return value
+	if (i == -1)
+	{
+		if (m_regs->PMODE.EN1 & m_regs->PMODE.EN2)
+		{
+			GSVector4i r[2] = { GetDisplayRect(0), GetDisplayRect(1) };
+			GSVector4i r_intersect = r[0].rintersect(r[1]);
+			GSVector4i r_union = r[0].runion_ordered(r[1]);
+
+			// If the conditions for passing the merged rectangle is unsatisfied, then
+			// pass the rectangle with the bigger size.
+			bool can_be_merged = !(r_intersect.width() && r_intersect.height()) || r_intersect.xyxy().eq(r_union.xyxy());
```

Comment:
Tell me if is the same as (or if I'm wrong)
```
!r_intersect.width() || !r_intersect.height() || r_intersect.xyxy().eq(r_union.xyxy())
```
If I'm right, don't you think it is easier this way ?